pilot/pkg/config/kube/crd/conversion.go

```diff
@@ -61,6 +62,38 @@ func ConvertObject(schema model.ProtoSchema, object IstioObject, domain string)
 	}, nil
 }
 
+// ConvertObject converts an IstioObject k8s-style object to the
+// internal configuration model.
+func ConvertObjectFromUnstructured(schema model.ProtoSchema, un *unstructured.Unstructured, domain string) (*model.Config, error) {
+	data, err := schema.FromJSONMap(un.Object["spec"])
+	if err != nil {
+		return nil, err
+	}
+
+	//// FIXME this is a gross hack to hardcode a service's domain name in kubernetes
```

Comment:
Do we know what would need to change to remove the hack?

Reply:
@rshriram added this [6 months ago](https://github.com/istio/istio/pull/4708/files#diff-853f0f8daedbb297899fab87fecec4d9R39) as part of https://github.com/istio/istio/pull/4708. 

We could remove the hack from Pilot to [expand short names to FQDN](https://github.com/istio/istio/pull/4708/files#diff-e3109392080297ee093b7189648289e1R568) if if we move the k8s service registry adapter out of pilot, e.g. into galley, or equiv (see https://github.com/istio/istio/issues/7698).

I don't think the cluster domain is needed for other Istio configuration.

Reply:
ah ok, thanks for the background.  Is there a tracking bug for this change? Maybe add a TODO with the link?

Reply:
the code is commented out. so can be removed fully.